sdk/maps/maps-weather-rest/src/mapsWeather.ts

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { ClientOptions } from "@azure-rest/core-client";
+import {
+  AzureKeyCredential,
+  AzureSASCredential,
+  isSASCredential,
+  isTokenCredential,
+  TokenCredential,
+} from "@azure/core-auth";
+import { bearerTokenAuthenticationPolicy } from "@azure/core-rest-pipeline";
+import { createMapsClientIdPolicy } from "@azure/maps-common";
+import { MapsWeatherClient } from "./generated";
+import createClient from "./generated/mapsWeatherClient";
+
+/**
+ * Creates an instance of MapsWeatherClient from a subscription key.
+ *
+ * @example
+ * ```ts
+ * import MapsWeather from "@azure-rest/maps-Weather";
+ *
+ * const credential = new AzureKeyCredential("<subscription-key>");
+ * const client = MapsWeather(credential);
+ *```
+ *
+ * @param credential - An AzureKeyCredential instance used to authenticate requests to the service
+ * @param options - Options used to configure the Weather Client
+ */
+export default function MapsWeather(
+  credential: AzureKeyCredential,
+  options?: ClientOptions,
+): MapsWeatherClient;
+/**
+ * Creates an instance of MapsWeather from an Azure Identity `TokenCredential`.
+ *
+ * @example
+ * ```ts
+ * import MapsWeatherClient from "@azure-rest/maps-weather";
+ * import { DefaultAzureCredential } from "@azure/identity";
+ *
+ * const credential = new DefaultAzureCredential();
+ * const client = MapsWeather(credential, "<maps-account-client-id>");
+ *```
+ *
+ * @param credential - An TokenCredential instance used to authenticate requests to the service
+ * @param mapsAccountClientId - The Azure Maps client id of a specific map resource
+ * @param options - Options used to configure the Weather Client
+ */
+export default function MapsWeather(
+  credential: TokenCredential,
+  mapsAccountClientId: string,
+  options?: ClientOptions,
+): MapsWeatherClient;
+/**
+ * Creates an instance of MapsWeather from an Azure Identity `AzureSASCredential`.
+ *
+ * @example
+ * ```ts
+ * import MapsWeather from "@azure-rest/maps-weather";
+ * import { AzureSASCredential } from "@azure/core-auth";
+ *
+ * const credential = new AzureSASCredential("<SAS Token>");
+ * const client = MapsWeather(credential);
+ * ```
+ *
+ * @param credential - An AzureSASCredential instance used to authenticate requests to the service
+ * @param options - Options used to configure the Weather Client
+ */
+export default function MapsWeather(
+  credential: AzureSASCredential,
+  options?: ClientOptions,
+): MapsWeatherClient;
+export default function MapsWeather(
+  credential: TokenCredential | AzureKeyCredential | AzureSASCredential,
+  clientIdOrOptions: string | ClientOptions = {},
+  maybeOptions: ClientOptions = {},
+): MapsWeatherClient {
+  const options = typeof clientIdOrOptions === "string" ? maybeOptions : clientIdOrOptions;
+
+  /**
+   * maps service requires a header "ms-x-client-id", which is different from the standard Microsoft Entra ID.
```

Comment:
the header should be "x-ms-client-id". All the comments are wrong in our other packages :(